presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java

```diff
@@ -792,7 +792,10 @@ private PhysicalOperation createMergeSource(RemoteSourceNode node, LocalExecutio
 
         private PhysicalOperation createRemoteSource(RemoteSourceNode node, LocalExecutionPlanContext context)
         {
-            if (!context.getDriverInstanceCount().isPresent()) {
+            if (node.isOrdered()) {
```

Comment:
I'm a little bit surprised here, so how does the `orderingScheme` being guaranteed before if we don't limit the driver instance? Are we doing another sort after exchange? If that's the case, then would this hurt those queries' performance since they don't need to have only a single remote source?

Reply:
The `orderingScheme` is used for distributed sort. When `orderingScheme` is present - the N way merge has to be done by the exchange operator. The "merging exchange" operator is always run with a concurrency equal to 1: (https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java#L756, https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java#L767)

Reply:
Maybe call this `ensureSourceOrdering` so reader won't confuse this `order` with `orderingScheme`.